apps/x509.c

```diff
@@ -604,10 +614,33 @@ int x509_main(int argc, char **argv)
     if (!opt_check_md(digest))
         goto opthelp;
 
+    if (preserve_dates && not_before != NULL) {
+        BIO_printf(bio_err, "Cannot use -preserve_dates with -not_before option\n");
+        goto err;
+    }
+    if (not_before != NULL && !ASN1_TIME_set_string_X509(NULL, not_before)) {
+        BIO_printf(bio_err,
+                   "start date is invalid, it should be YYMMDDHHMMSSZ or YYYYMMDDHHMMSSZ\n");
+        goto err;
+    }
```

Comment:
Well, `ASN1_TIME_set_string_X509()` will be called anyway by `set_cert_times()` called below, but I see the point of providing a specific error message.